### PR TITLE
feat(mobile): quiz screen bb-245

### PR DESCRIPTION
**Pull request by KotenkoNata**
- I created the quiz screen that leverages the reusable RadioGroup component to display multiple-choice answers.
- Implemented a counter to indicate quiz progress visually.
- Added navigation logic to transition to the "Analyzing" screen upon selecting an answer for the last question.

[Quiz_screen.webm](https://github.com/user-attachments/assets/04221327-c26f-43e0-9670-b68d3fee288d)

### CODE REVIEW

File: apps/mobile/src/packages/quiz/quiz-api.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import { QuizApiPath } from "./libs/enums/enums";
+import { type QuizQuestionDto } from "./libs/types/types";
+
+type Constructor = {
+	baseUrl: string;
+	http: HTTP;
+	storage: Storage;
+};
+
```
Comment (anogire) [inline apps/mobile/src/packages/quiz/quiz-api.ts:14]: Let's move this type to libs/types/api or even to shared as suggested @katerynakostikova

File: apps/mobile/src/screens/quiz/libs/components/counter/styles.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { StyleSheet } from "react-native";
+
+import { BaseColor } from "~/libs/enums/enums";
+
+const styles = StyleSheet.create({
+	container: {
+		alignSelf: "center",
```
Comment (revenets) [inline apps/mobile/src/screens/quiz/libs/components/counter/styles.ts:7]: Don't we have this in globalStyles?

File: apps/mobile/src/packages/quiz/quiz-api.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { type Storage } from "~/libs/packages/storage/storage";
+
+import { QuizApiPath } from "./libs/enums/enums";
+import { type QuizQuestionDto } from "./libs/types/types";
+
+type Constructor = {
+	baseUrl: string;
+	http: HTTP;
+	storage: Storage;
+};
```
Comment (revenets) [inline apps/mobile/src/packages/quiz/quiz-api.ts:13]: Let's make this improvement and move `Constructor` type to shared folder, please

**Review by revenets — CHANGES_REQUESTED** [review]

File: apps/mobile/src/libs/components/infinite-pager/infinite-pager.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import { PREVIOUS_INDEX_OFFSET } from "~/libs/constants/constants";
+import { type InfinitePagerImperativeApi } from "~/libs/types/types";
+
+import { getPageInterpolatorSlide } from "./libs/animations/animations";
+import { ANIMATION_CONFIG } from "./libs/constants/constants";
+
+type Properties = {
+	infinitePagerReference: React.MutableRefObject<InfinitePagerImperativeApi | null>;
+	renderPageComponent: () => JSX.Element;
```
Comment (fshabanov) [inline apps/mobile/src/libs/components/infinite-pager/infinite-pager.tsx:12]: A9

File: apps/mobile/src/libs/types/types.ts [hunk 5]
```diff
@@ -1,5 +1,6 @@
 export { type AsyncThunkConfig } from "./app/app";
 export { type IconName } from "./icon/icon";
+export { type InfinitePagerImperativeApi } from "./infinite-pager/types";
```
Comment (fshabanov) [inline apps/mobile/src/libs/types/types.ts:3]:
```suggestion
export { type InfinitePagerImperativeApi } from "./infinite-pager/infinite-pager";
```

File: apps/mobile/src/libs/components/infinite-pager/infinite-pager.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const InfinitePager = ({
+	infinitePagerReference,
+	renderPageComponent,
+}: Properties): JSX.Element => {
+	return (
+		<RNInfinitePager
+			animationConfig={ANIMATION_CONFIG}
+			gesturesDisabled
+			pageBuffer={PREVIOUS_INDEX_OFFSET}
```
[the remark this answers is not in the document]
Comment (revenets) [reply apps/mobile/src/libs/components/infinite-pager/infinite-pager.tsx:23]: `pageBuffer` relates to the number of pages to render on either side of the active page, so `PREVIOUS_INDEX_OFFSET` doesn't seem like the right name for the variable here.

[the remark this answers is not in the document]
Comment (fshabanov) [reply apps/mobile/src/libs/components/infinite-pager/infinite-pager.tsx:23]: You're right, misunderstood the purpose of pageBuffer

File: apps/mobile/src/screens/quiz/libs/components/counter/styles.ts (hunk 2 above)
Comment (KotenkoNata) [reply apps/mobile/src/screens/quiz/libs/components/counter/styles.ts:7]: No, we have  - alignItems.

File: apps/mobile/src/packages/quiz/quiz-api.ts (hunk 3 above)
Comment (KotenkoNata) [reply apps/mobile/src/packages/quiz/quiz-api.ts]: Done

**Review by anogire — APPROVED** [review]

**Review by revenets — APPROVED** [review]

File: apps/mobile/src/packages/quiz/quiz-api.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import { APIPath, ContentType } from "~/libs/enums/enums";
+import { type ApiConfiguration, BaseHttpApi } from "~/libs/packages/api/api";
```
Comment (fshabanov) [inline apps/mobile/src/packages/quiz/quiz-api.ts:2]:
```suggestion
import { type APIConfiguration, BaseHttpApi } from "~/libs/packages/api/api";
```
B23

File: apps/mobile/src/screens/quiz/quiz.tsx [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+	const handleFormSubmit = useCallback((): void => {
+		void handleSubmit(handleNextClick)();
+
+		if (isLastQuestion) {
+			navigation.navigate(RootScreenName.WHEEL_LOADING);
+		}
+	}, [handleNextClick, handleSubmit, isLastQuestion, navigation]);
+
+	const renderPageComponent = useCallback(() => {
```
Comment (fshabanov) [inline apps/mobile/src/screens/quiz/quiz.tsx:103]:
```suggestion
	const handleRenderPageComponent = useCallback(() => {
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/quiz/quiz.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+	const {
+		answersByQuestionIndex,
+		currentQuestion,
+		currentQuestionIndex,
+		dataStatus,
+		questions,
+	} = useAppSelector((state) => state.quiz);
+
+	const currentAnswer = answersByQuestionIndex[currentQuestionIndex] ?? "";
+	const totalQuestionsAmount = questions.length - PREVIOUS_INDEX_OFFSET;
```
Comment (fshabanov) [inline apps/mobile/src/screens/quiz/quiz.tsx:55]:
```suggestion
	const totalQuestionsAmount = questions.length;
```
Why do you subtract PREVIOUS_INDEX_OFFSET?

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/quiz/quiz.tsx (hunk 9 above)
Comment (KotenkoNata) [reply apps/mobile/src/screens/quiz/quiz.tsx:55]: I misunderstood and reverted the changes I made.

**Review by WilhelmWesser — APPROVED** [review]

File: apps/mobile/src/screens/quiz/quiz.tsx (hunk 9 above)
Comment (fshabanov) [reply apps/mobile/src/screens/quiz/quiz.tsx:55]: Not fixed

File: apps/mobile/src/screens/quiz/quiz.tsx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+								<Text
+									preset="subheading"
+									style={[globalStyles.mb24, styles.title]}
+									weight="bold"
+								>
+									Wheel Quiz questions
+								</Text>
+								<Counter
+									currentStep={currentQuestionIndex + NumericalValue.ONE}
+									totalSteps={totalQuestionsAmount + NumericalValue.ONE}
```
Comment (fshabanov) [inline apps/mobile/src/screens/quiz/quiz.tsx:135]:
```suggestion
									totalSteps={totalQuestionsAmount}
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**